### PR TITLE
Map display fixes, cop pathfinding fix, GAMESND refactoring

### PR DESCRIPTION
**Pull request by SoapyMan**

### CODE REVIEW

File: src_rebuild/GAME/C/GAMESND.C [hunk 1]
```diff
@@ -594,6 +616,10 @@ void LoadLevelSFX(int missionNum)
 			i++;
 		} while (i < 3);
```
Comment (Fireboyd78) [inline src_rebuild/GAME/C/GAMESND.C:617]: Please `for` the love of god..

File: src_rebuild/GAME/C/ENVSOUND.C [hunk 2]
```diff
@@ -28,6 +28,26 @@ void AddEnvSounds(int level, int time)
...
+		case 1:
+			AddEnvSnd(3, 0, 4, 2, 2000, -0x38ba8, -0x41096, -0x38ba8, -0x41096);
+			AddEnvSnd(3, 0, 4, 2, 2000, 0x3d928, -0x1d2cc, 0x3d928, -0x1d2cc);
+		case 5:
+			AddEnvSnd(1, 0, 4, 4, -0xdac, 0x4f236, -0x58dae, 0x52e2c, -0x4ee80);
+			AddEnvSnd(1, 0, 4, 4, -0xdac, 0x52e2c, -0x4fbc8, 0x56734, -0x51338);
+
+			if (level == 5)
+			{
+				AddEnvSnd(1, 0x10, 4, 3, -0x1194, -0x1bcec, -0x2c308, -0x14244, -250000);
```
Comment (Fireboyd78) [inline src_rebuild/GAME/C/ENVSOUND.C:40]: Should convert these to decimal, just hover over the number in Visual Studio.

**Review by Fireboyd78 — APPROVED** [review]
I'm sure it works just fine, most of my comments are about aesthetics/brevity/refactoring.

File: src_rebuild/GAME/C/GAMESND.C [hunk 3]
```diff
@@ -1473,531 +1473,649 @@ char force_siren[8] = { 0 };
...
+			indexlist[num_noisy_cars] = i;
+			car_dist[i] = dist;
+			num_noisy_cars++;
+		}
+	}
 
-		if (pitch < 0x4000)
+	// sort cars by distance distance
+	i = 0;
+	while (i < num_noisy_cars-1)
```
[the remark this answers is not in the document]
Comment (SoapyMan) [reply src_rebuild/GAME/C/GAMESND.C]: is use fors when needed to continue :D

File: src_rebuild/GAME/C/ENVSOUND.C (hunk 2 above)
Comment (SoapyMan) [reply src_rebuild/GAME/C/ENVSOUND.C:40]: sadly we have only a VS code plugin

File: src_rebuild/GAME/C/GAMESND.C [hunk 4]
```diff
@@ -1473,531 +1473,649 @@ char force_siren[8] = { 0 };
...
-					uVar3 = *puVar15;
-					if (car_dist[*puVar7] < car_dist[uVar3])
-					{
-						*puVar15 = *puVar7;
-						*puVar7 = uVar3;
-					}
-					vvar4--;
-					puVar7++;
-				} while (vvar4 != 0);
+				if (gCurrentMissionNumber == 26 || car_data[car].controlType != CONTROL_TYPE_CIV_AI)
```
[the remark this answers is not in the document]
Comment (SoapyMan) [reply src_rebuild/GAME/C/GAMESND.C:1652]: good point

**Pull request merged**